sympy/functions/elementary/hyperbolic.py

```diff
@@ -156,12 +156,13 @@ def eval(cls, arg):
                 x = arg.args[0]
                 return 1/(sqrt(x - 1) * sqrt(x + 1))
 
-    @staticmethod
     @cacheit
-    def taylor_term(n, x, *previous_terms):
+    def taylor_term(self, n, x, *previous_terms):
         """
         Returns the next term in the Taylor series expansion.
         """
+        if self.args[0] != x:
+            return super(sinh, self).taylor_term(n, x, *previous_terms)
```

Comment:
According the comment in the issue, for Py 3 this can be `super().taylor_term(...` here and below.